### PR TITLE
[WIP]Add container runtime driver

### PR DESCRIPTION
**Pull request by kakaZhou719**
### Describe what this PR does / why we need it

Add container runtime driver:  support to config container runtime, like access private registry.

### CODE REVIEW

File: pkg/container-runtime/driver.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+func (d DockerRuntimeDriver) configRegistryCert(domain, port string, caFile string, ips []net.IP) error {
+	// copy ca cert to "/etc/docker/certs.d/${domain}:${port}/
+}
+
+func (d DockerRuntimeDriver) configDaemon(endpoint string, ips []net.IP) error {
+	// modify daemon.json: "mirrors": ["${endpoint}"],docker server need to restart if modify daemon.json
+}
+
+func NewDockerRuntimeDriver(ssh infradriver.InfraDriver) ContainerRuntimeDriver {
```
Comment (VinceCui) [inline pkg/container-runtime/driver.go:72]: Don't name as "ssh"

File: pkg/container-runtime/driver.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+}
+
+//ContainerRuntimeDriver provide configuration Interface for different container runtime.
+type ContainerRuntimeDriver interface {
+	// ConfigRegistry config registry via each container runtime ip address
+	ConfigRegistry(ConfigForRegistry, []net.IP) error
+}
+
+type DockerRuntimeDriver struct {
+	ssh infradriver.InfraDriver
```
Comment (VinceCui) [inline pkg/container-runtime/driver.go:45]: Too

File: pkg/container-runtime/driver.go [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+// limitations under the License.
+
+package container_runtime
+
+import (
+	"github.com/sealerio/sealer/pkg/infradriver"
+	"net"
+)
+
+type ConfigForRegistry struct {
```
Comment (VinceCui) [inline pkg/container-runtime/driver.go:22]: Why should ContainerRuntimeInstaller know ConfigForRegistry?

Comment (kakaZhou719) [reply pkg/container-runtime/driver.go:22]: this is not for `ContainerRuntimeInstaller`, its `Configurator` for ContainerRuntime, for example: config ContainerRuntime to access local registry.

**Pull request closed without merging**